lib/active_merchant/billing/gateways/paymentez.rb

```diff
@@ -134,7 +134,8 @@ def add_customer_data(post, options)
         post[:user][:email] = options[:email]
         post[:user][:ip_address] = options[:ip] if options[:ip]
         post[:user][:fiscal_number] = options[:fiscal_number] if options[:fiscal_number]
-        post[:user][:phone] = options[:phone] || (options[:billing_address][:phone] if options[:billing_address])
+        post[:user][:phone] = options[:phone] || (options[:billing_address][:phone] if options[:billing_address] &&
+          options[:billing_address][:phone])
```

Comment:
Nice, I thought that might be it as well. A couple small things that probably aren't worth changing for this:
- This is a good use case for the `.dig` method (new to the recently updated ruby version) https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig
- Usually we don't bother sticking to the 80 characters per line in adapters, but NBD.